app/components/UserAttendance/GroupFilter.tsx

```diff
@@ -0,0 +1,82 @@
+import { Flex } from '@webkom/lego-bricks';
+import { useMemo, useState } from 'react';
+import Pill from 'app/components/Pill';
+import styles from 'app/components/UserAttendance/AttendanceModalContent.module.css';
+import { selectCurrentUser } from 'app/reducers/auth';
+import { selectAllGroups } from 'app/reducers/groups';
+import { useAppSelector } from 'app/store/hooks';
+import type { EntityId } from '@reduxjs/toolkit';
+import type { AttendanceModalRegistration } from 'app/components/UserAttendance/AttendanceModalContent';
+
+const excludeFilterGroups: EntityId[] = [
+  1, // Users
+  2, // Abakus
+  47, // Formaterte
+];
+
+type Props = {
+  registrations: AttendanceModalRegistration[];
+  groupFilter: EntityId | null;
+  setGroupFilter: (group: EntityId | null) => void;
+};
+
+export const GroupFilter = ({
+  registrations,
+  groupFilter,
+  setGroupFilter,
+}: Props) => {
+  const currentUser = useAppSelector(selectCurrentUser);
+  const allGroups = useAppSelector(selectAllGroups);
+  const [hovered, setHovered] = useState(false);
+  const expanded = hovered || groupFilter !== null;
+
+  const topGroups = useMemo(
+    () =>
+      allGroups
+        .filter(
+          (group) =>
+            currentUser?.abakusGroups.includes(group.id) &&
+            !excludeFilterGroups.includes(group.id),
+        )
+        .map(
+          (group) =>
+            [
+              group,
+              registrations.filter(
+                (reg) =>
+                  'abakusGroups' in reg.user &&
+                  reg.user.abakusGroups.includes(group.id),
+              ).length,
+            ] as const,
+        )
+        .sort((a, b) => b[1] - a[1]),
+    [allGroups, currentUser?.abakusGroups, registrations],
+  );
+
+  return (
+    <Flex
+      wrap
+      gap={6}
+      style={{ maxHeight: expanded ? undefined : 50 }}
+      className={styles.groupFilters}
+      onMouseEnter={() => setHovered(true)}
+      onMouseLeave={() => setHovered(false)}
+    >
+      {topGroups.map(([group]) => (
+        <Pill
+          key={group.id}
+          onClick={() =>
+            groupFilter === group.id
+              ? setGroupFilter(null)
+              : setGroupFilter(group.id)
+          }
+          className={styles.groupFilterButton}
+          color={groupFilter === group.id ? 'var(--color-blue-4)' : undefined}
```

Comment:
Not sure about the blue color since that kind of unmatches the rest of the webapp. Would be nice to do something similar as the group selection thingy in the event editor:
<img width="380" alt="image" src="https://github.com/user-attachments/assets/e81df971-c2c0-411b-ab1d-114e705479bc" />